cli/command/cli.go

```diff
@@ -268,8 +279,8 @@ type ClientInfo struct {
 }
 
 // NewDockerCli returns a DockerCli instance with IO output and error streams set by in, out and err.
-func NewDockerCli(in io.ReadCloser, out, err io.Writer, isTrusted bool, containerizedFn func(string) (clitypes.ContainerizedClient, error)) *DockerCli {
```

Comment:
Is this change really needed? I'm afraid it might break a lots of downstream projects... 😅

Reply:
It's worth mentioning that this is not the same interface as `github.com/docker/docker/client` which is the "programming API client" we'd expect those wanting to speak to the Docker API to be using and which includes flexibility wrt doing trust stuff.

This here is a (pretty much CLI internal) interface which is part of the docker/cli infrastructure/scaffolding, the consumers of this content trust setting are all in `docker/cli/cli/**/*.go` (callers of `dockerCli.ContentTrustEnabled()`). We do have some external users of this CLI scaffolding I think  -- but they are prime candidates for becoming plugins since their reason for using the docker/cli scaffolding is in essence to be compatible with the cli in some ways.

In any case it seems wrong to allow alternative CLI tools consuming the `docker/cli/cli/command` infrastructure to opt into or out of obeying the user's `$DOCKER_CONTENT_TRUST` setting.

Even if we do think it is a reasonable knob to give consumers I don't think making them manually decode `$DOCKER_CONTENT_TRUST` is the right way to approach it.

FWIW this was done in c15597947233a4a151dc5fa286e324c6ee0a6b4a:
```
    Push handling of `$DOCKER_CONTENT_TRUST` down into `NewDockerCli`
    
    I'm about to add a new (internal) user of this function and this saves
    exporting or duplicating then logic.
    
    Both the `docs/yaml` and `man` use don't do anything which would be affected by
    enabling/disabling content trust (nor would you expect them to) so switching
    from a hardcoded `false` to probing the envvar makes no practicaly difference.
    
    Note that this changes the API of `NewDockerCli`, but actual external callers
    should be unexpected.
    
    Signed-off-by: Ian Campbell <ijc@docker.com>
```

Reply:
Ok I'm fine with the explanation, and you are right my concerns were about docker-app, which will be translated as a plugin. Any other concerns about that cc @thaJeztah @vdemeester ?

Reply:
I'm going to be handling this by rebasing onto #1633.